### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by SuperQ**
* Update to Go modules.
* Update all vendoring.
* Fix up prometheus client_golang use.
* Convert from Travis to CircleCI build pipeline.
* Update promu.yml
* Update github repo locations.
* Use Prometheus common Makefile.
* Add missing license headers.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+version: 2.1
+
+orbs:
+  prometheus: prometheus/prometheus@0.1.0
```
Comment (simonpasquier) [inline .circleci/config.yml]: it could be  `prometheus/prometheus@0.4.0`

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  codespell:
+    docker:
+    - image: circleci/python
+
+    steps:
+    - checkout
+    - run: sudo pip install codespell
+    - run: codespell --skip=".git,./vendor,ttar,go.mod,go.sum,*pem" -L uint,packages\',uptodate
+
+  build:
```
Comment (simonpasquier) [inline .circleci/config.yml]: Can't you use the `build` job from the CircleCI's orb as [here](https://github.com/prometheus/prometheus/blob/000e306f2fbb373bc37fa8e710553b072fac71a1/.circleci/config.yml#L97-L101)?

File: .circleci/config.yml (hunk 1 above)
Comment (SuperQ) [reply .circleci/config.yml]: Ahh, I just copy-pasta'd from node_exporter. I guess we need to sync these.

We should bump the major version to `1.0.0`, then we can use this syntax:
```yaml
orbs:
  prometheus: prometheus/prometheus@1
```

File: .circleci/config.yml (hunk 2 above)
Comment (SuperQ) [reply .circleci/config.yml]: Sure, again, copy-pasta from node_exporter.

**Comment by SuperQ** [comment]
I also realized that we need to include the `prometheus-communit` docker repo override.

**Pull request merged**